src/dev/lupio/lupio_tmr.hh

```diff
@@ -67,15 +65,17 @@ class LupioTMR : public BasicPioDevice
         LUPIO_TMR_MAX,
     };
 
-    // Timer registers
-    uint64_t reload = 0;
-
-    // Control
-    bool ie = false;
-    bool pd = false;
+    struct LupioTimer
+    {
+        Event *tmrEvent = nullptr;
+        uint64_t reload = 0;
+        bool ie = false;            // Control
+        bool pd = false;
+        bool expired = false;       // Status
+        Tick startTime = 0;
+    };
 
-    // Status
-    bool expired = false;
+    std::vector<LupioTimer> _timers;
```

Comment:
Why `_timers` and not `timers`? You don't use leading underscores for private members anywhere else.